x/bank/types/genesis.go

```diff
@@ -63,6 +63,18 @@ func (gs GenesisState) Validate() error {
 
 // NewGenesisState creates a new genesis state.
 func NewGenesisState(params Params, balances []Balance, supply sdk.Coins,
+	denomMetaData []Metadata) *GenesisState {
+	return &GenesisState{
+		Params:        params,
+		Balances:      balances,
+		Supply:        supply,
+		DenomMetadata: denomMetaData,
+		SupplyOffsets: []GenesisSupplyOffset{},
+	}
+}
+
+// NewGenesisState creates a new genesis state with supply offsets
+func NewGenesisStateWithSupplyOffsets(params Params, balances []Balance, supply sdk.Coins,
```

Comment:
nit: we can also do a decorator pattern style like here with gov deposit params:
https://github.com/osmosis-labs/cosmos-sdk/blob/c4f55a31756ff8bf63c9c17be75fb5f85c1d580e/x/gov/types/params.go#L94